@xen-orchestra/lite/src/components/menu/AppMenu.vue

```diff
@@ -0,0 +1,91 @@
+<template>
+  <slot :is-open="isOpen" :open="open" name="trigger" />
+  <Teleport to="body" :disabled="!isRoot || !slots.trigger">
+    <ul
+      v-if="!$slots.trigger || isOpen"
+      ref="menu"
+      :class="{ horizontal, shadow }"
+      class="app-menu"
+      v-bind="$attrs"
+    >
+      <slot />
+    </ul>
+  </Teleport>
+</template>
+
+<script lang="ts" setup>
+import placement, { type Options } from "placement.js";
+import { inject, nextTick, provide, ref, toRef, unref, useSlots } from "vue";
+import { onClickOutside, unrefElement, whenever } from "@vueuse/core";
+
+const props = defineProps<{
+  horizontal?: boolean;
+  shadow?: boolean;
+  disabled?: boolean;
+  placement?: Options["placement"];
+}>();
+const isRoot = inject("isMenuRoot", true);
+provide("isMenuRoot", false);
+const slots = useSlots();
+const isOpen = ref(false);
+const menu = ref();
+const isParentHorizontal = inject("isMenuHorizontal", undefined);
+provide("isMenuHorizontal", toRef(props, "horizontal"));
```

Comment:
Could you explain this a bit more? You're expecting an optional `placement` prop to force the placement and then another `horizontal` prop *and* a provided `isMenuHorizontal` to determine where to put the menu. What's the normal use case for all of those? Who's supposed to provide what in a basic situation?

Reply:
The `placement` prop is where the menu should appear relative to the trigger.

The `horizontal` prop is to set the layout of the menu itself.

If `true`, the items will be arranged from left to right. If `false` they will be arranged from top to bottom.

The `provide` is here to pass the info to its children. It is used in `MenuItem` to know which "submenu icon" to show.

If the menu is horizontal, then the icon will be an "angle-down". Otherwise it will be an "angle-right"

![Capture 2022-09-02 at 09 27 54@2x](https://user-images.githubusercontent.com/19408/188082813-98f2161b-e341-40f3-aaba-399e14a812a2.png)

The normal use case is just to set the `horizontal` prop on the main `AppMenu` if needed.

The `placement` prop would be used for specific needs only.

Reply:
Thanks :)